components/asana/asana.app.js

```diff
@@ -0,0 +1,251 @@
+const axios = require("axios");
+const crypto = require("crypto");
+
+module.exports = {
+  type: "app",
+  app: "asana",
+  propDefinitions: {
+    workspaceId: {
+      type: "string",
+      label: "Workspace",
+      optional: false,
+      async options(opts) {
+        const workspaces = await this.getWorkspaces();
+        return workspaces.map((workspace) => {
+          return {
+            label: workspace.name,
+            value: workspace.gid,
+          };
+        });
+      },
+    },
+    projectId: {
+      type: "string",
+      label: "Project",
+      optional: false,
+      async options(opts) {
+        const projects = await this.getProjects(opts.workspaceId);
+        return projects.map((project) => {
+          return {
+            label: project.name,
+            value: project.gid,
+          };
+        });
+      },
+    },
+    taskIds: {
+      type: "string[]",
+      label: "Tasks",
+      async options(opts) {
+        const tasks = await this.getTasks(opts.projectId);
+        return tasks.map((task) => {
+          return { label: task.name, value: task.gid };
+        });
+      },
+    },
+    organizationId: {
+      type: "string",
+      label: "Organization",
+      optional: false,
+      async options(opts) {
+        const organizations = await this.getOrganizations();
+        return organizations.map((organization) => {
+          return {
+            label: organization.name,
+            value: organization.gid,
+          };
+        });
+      },
+    },
+  },
+  methods: {
+    async _getHeaders() {
+      return {
+        Accept: "application/json",
+        Authorization: `Bearer ${this.$auth.oauth_access_token}`,
+      };
+    },
+    async _getAuthorizationHeader({ data, method, url, headers }) {
+      const token = {
+        key: this.$auth.oauth_access_token,
+        secret: this.$auth.oauth_refresh_token,
+      };
+      return await axios({
+        method: "POST",
+        url: `https://app.asana.com/api/1.0/webhooks`,
+        data: data.body,
+        headers,
+      });
+    },
+    async _makeWebhookRequest(config) {
+      if (!config.headers) config.headers = {};
+      const authorization = await this._getAuthorizationHeader(config);
+      config.headers.authorization = authorization;
+      try {
+        await axios(config);
+      } catch (err) {
+        console.log(err);
+      }
+      return authorization.data;
+    },
+    async createHook(body) {
+      const resp = await this._makeWebhookRequest({
+        method: "post",
+        url: `https://app.asana.com/api/1.0/webhooks`,
```

Comment:
Can you move some of this logic to the `_makeWebhookRequest` method itself, since each webhook request will use the same URL and headers?